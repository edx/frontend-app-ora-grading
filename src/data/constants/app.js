import { getConfig } from '@edx/frontend-platform';

export const routePath = `${getConfig().PUBLIC_PATH}:courseId`;
export const locationId = window.location.pathname.slice(1);
