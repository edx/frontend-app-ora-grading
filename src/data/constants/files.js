import { StrictDict } from 'utils';

export const FileTypes = StrictDict({
  pdf: 'pdf',
  jpg: 'jpg',
  jpeg: 'jpeg',
  png: 'png',
  bmp: 'bmp',
});

export default FileTypes;
