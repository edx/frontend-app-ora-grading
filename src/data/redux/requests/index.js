export { actions, reducer } from './reducer';
