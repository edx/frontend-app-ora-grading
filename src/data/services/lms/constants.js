import { StrictDict } from 'utils';

export const lockStatuses = StrictDict({
  unlocked: 'unlocked',
  locked: 'locked',
  inProgress: 'in-progress',
});

export const gradeStatuses = StrictDict({
  ungraded: 'ungraded',
  graded: 'graded',
});

export const gradingStatuses = StrictDict({
  ungraded: gradeStatuses.ungraded,
  graded: gradeStatuses.graded,
  locked: lockStatuses.locked,
  inProgress: lockStatuses.inProgress,
});

export const gradingStatusDisplay = StrictDict({
  [gradingStatuses.ungraded]: 'Ungraded',
  [gradingStatuses.locked]: 'Currently being graded by someone else',
  [gradingStatuses.graded]: 'Grading Complete',
  [gradingStatuses.inProgress]: 'You are currently grading this response',
});

export const feedbackRequirement = StrictDict({
  disabled: 'disabled',
  required: 'required',
  optional: 'optional',
});
