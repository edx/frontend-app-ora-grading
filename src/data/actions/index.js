import { StrictDict } from 'utils';

import app from './app';
import grading from './grading';
import submissions from './submissions';

export default StrictDict({
  app,
  grading,
  submissions,
});
