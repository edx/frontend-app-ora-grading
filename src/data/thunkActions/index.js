import { StrictDict } from 'utils';

import app from './app';
import grading from './grading';

export default StrictDict({
  app,
  grading,
});
