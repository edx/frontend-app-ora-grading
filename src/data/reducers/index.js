import { combineReducers } from 'redux';

import app from './app';
import submissions from './submissions';

/* istanbul ignore next */
const rootReducer = combineReducers({
  app,
  submissions,
});

export default rootReducer;
