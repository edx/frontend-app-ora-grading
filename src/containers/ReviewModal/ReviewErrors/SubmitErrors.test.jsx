import React from 'react';
import { shallow } from 'enzyme';

import { actions, selectors, thunkActions } from 'data/redux';
import { ErrorStatuses, RequestKeys } from 'data/constants/requests';

import {
  SubmitErrors,
  mapStateToProps,
  mapDispatchToProps,
} from './SubmitErrors';

jest.mock('data/redux', () => ({
  actions: {
    requests: {
      clearRequest: jest.fn().mockName('actions.requests.clearRequest'),
    },
  },
  selectors: {
    requests: {
      errorStatus: (...args) => ({ errorStatus: args }),
    },
  },
  thunkActions: {
    grading: {
      submitGrade: jest.fn().mockName('thunkActions.grading.submitGrade'),
    },
  },
}));

let el;
jest.mock('./ReviewError', () => 'ReviewError');

const requestKey = RequestKeys.submitGrade;

describe('SubmitErrors component', () => {
  const props = {};
  describe('component', () => {
    beforeEach(() => {
      props.resubmit = jest.fn();
      props.clearRequest = jest.fn();
      el = shallow(<SubmitErrors {...props} />);
      el.instance().dismissError = jest.fn().mockName('this.dismissError');
    });
    describe('snapshots', () => {
      test('snapshot: no failure', () => {
        expect(el.instance().render()).toMatchSnapshot();
      });
      test('snapshot: with network failure', () => {
        el.setProps({ errorStatus: ErrorStatuses.badRequest });
        expect(el.instance().render()).toMatchSnapshot();
      });
      test('snapshot: with conflict failure', () => {
        el.setProps({ errorStatus: ErrorStatuses.conflict });
        expect(el.instance().render()).toMatchSnapshot();
      });
    });
  });
  describe('mapStateToProps', () => {
    let mapped;
    const testState = { some: 'test-state' };
    beforeEach(() => {
      mapped = mapStateToProps(testState);
    });
    test('errorStatus loads from requests.errorStatus(fetchSubmission)', () => {
      expect(mapped.errorStatus).toEqual(
        selectors.requests.errorStatus(testState, { requestKey }),
      );
    });
  });
  describe('mapDispatchToProps', () => {
    it('loads clearRequest from actions.requests.clearRequest', () => {
      expect(mapDispatchToProps.clearRequest).toEqual(actions.requests.clearRequest);
    });
    it('loads resubmit from thunkActions.grading.submitGrade', () => {
      expect(mapDispatchToProps.resubmit).toEqual(thunkActions.grading.submitGrade);
    });
  });
});
