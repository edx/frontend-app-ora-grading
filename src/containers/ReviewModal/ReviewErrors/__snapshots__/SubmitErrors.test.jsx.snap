// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitErrors component component snapshots snapshot: no failure 1`] = `null`;

exports[`SubmitErrors component component snapshots snapshot: with conflict failure 1`] = `
<ReviewError
  actions={
    Object {
      "cancel": Object {
        "message": Object {
          "defaultMessage": "Dismiss",
          "description": "Dismiss error action button text",
          "id": "ora-grading.ReviewModal.dismiss",
        },
        "onClick": [MockFunction this.dismissError],
      },
      "confirm": undefined,
    }
  }
  headingMessage={
    Object {
      "defaultMessage": "Error submitting grade",
      "description": "Error Submitting Grade heading text",
      "id": "ora-grading.ReviewModal.errorSubmittingGrade.Heading",
    }
  }
>
  <FormattedMessage
    defaultMessage="It looks like someone else got here first!  Your grade submission has been rejected"
    description="Error Submitting Grade content"
    id="ora-grading.ReviewModal.errorSubmittingGrade.Content"
  />
</ReviewError>
`;

exports[`SubmitErrors component component snapshots snapshot: with network failure 1`] = `
<ReviewError
  actions={
    Object {
      "cancel": Object {
        "message": Object {
          "defaultMessage": "Dismiss",
          "description": "Dismiss error action button text",
          "id": "ora-grading.ReviewModal.dismiss",
        },
        "onClick": [MockFunction this.dismissError],
      },
      "confirm": Object {
        "message": Object {
          "defaultMessage": "Resubmit grate",
          "description": "Resubmit grade button after network failure",
          "id": "ora-grading.ReviewModal.resubmitGrade",
        },
        "onClick": [MockFunction],
      },
    }
  }
  headingMessage={
    Object {
      "defaultMessage": "Grade not submitted",
      "description": "Grade submission network error heading",
      "id": "ora-grading.ReviewModal.gradeNotSubmitted.heading",
    }
  }
>
  <FormattedMessage
    defaultMessage="We're sorry, something went wrong when we tried to submit this grade.  Please try again."
    description="Grade submission network error message"
    id="ora-grading.ReviewModal.gradeNotSubmitted.heading"
  />
</ReviewError>
`;
